packages/tracking_repository/lib/src/tracking_repository.dart

```diff
@@ -21,8 +21,85 @@ class TrackingRepository {
   void track(TrackEvent event) {
     _firebase.logEvent(name: event.name);    
     dev.log(event.name);
-    // firebase
-    // GA
-    // tracker?.track...
+  }
+
+  /// Updates tracking services accordingly
+  void trackPoolCreation(TrackEvent event, 
+    String valueOne,
+    String valueTwo,
+    String lpTokens,
+    String shareOfPool,
+    String lpTokenName,
+    String walletId,
+  ) {
+    _firebase.logEvent(name: event.name,
+      parameters: {
+        'value_1' : valueOne,
```

Comment:
All these Events should be created on the BL side. The tracking_repository should not have any knowledge on what are the business requirements of "building events".
It forwards the events received to different services.

After calling the cubit method of "onPoolCreated" (the event should be created there and only call `trackingRepository.track(event)`